blurup.js

```diff
@@ -61,21 +73,29 @@ export async function createBlurUp(playbackId, options) {
         `[@mux/blurup] Error fetching thumbnail. 403: Forbidden. This Playback ID may require a thumbnail token. See https://docs.mux.com/guides/video/secure-video-playback for more information.`
       );
     }
-  } else if (response.status >= 400) {
+  } else if (response.status >= 400 ) {
     throw new Error(
       `[@mux/blurhash] Error fetching thumbnail. ${response.status}: ${response.statusText}`
     );
+  } else if (sourceResponse.status >= 400 ) {
+    throw new Error(
+      `[@mux/blurhash] Error fetching thumbnail. ${sourceResponse.status}: ${sourceResponse.statusText}`
+    );
   }
 
-  const arrayBuffer = await response.arrayBuffer();
+  const [arrayBuffer, sourceArrayBuffer] = await Promise.all([response.arrayBuffer(), sourceResponse.arrayBuffer()]);
```

Comment:
for the sourceResponse we don't want to wait and download the whole thing. there's feature in image-dimensions to get the image size from a stream https://github.com/sindresorhus/image-dimensions#usage
since we don't care about the source image data.